webapp/src/hook/hook.js

```diff
@@ -37,26 +41,55 @@ export default class Hooks {
             if (commands.length > 2) {
                 return Promise.resolve({error: {message: 'Too many arguments'}});
             }
-            privateKey = Buffer.from(commands[1], 'base64');
-            publicKey = getPublicKeyFromPrivateKey(privateKey);
-            if (!publicKey) {
+
+            key = keyFromString(commands[1]);
```

Comment:
let's use specific names for the key variables. Use privateKey or publicKey for more readability